### ISSUE TITLE
`eval` is used, which is unsafe

### ISSUE THREAD

**Issue by eric-wieser** [ros-perception/calibration#40]
In the following places:
- https://github.com/ros-perception/calibration/blob/20d3bbf5ec78027831f75cf90c2aac737a00e978/calibration_estimation/src/calibration_estimation/single_transform.py#L48
- https://github.com/ros-perception/calibration/blob/20d3bbf5ec78027831f75cf90c2aac737a00e978/calibration_estimation/src/calibration_estimation/joint_chain.py#L52
- https://github.com/ros-perception/calibration/blob/20d3bbf5ec78027831f75cf90c2aac737a00e978/calibration_estimation/src/calibration_estimation/urdf_params.py#L144

What type of input is this trying to parse?

**Comment by eric-wieser** [comment]
All the cases seem to be `eval(str(x))`. Is this:
- Trying to be `copy.copy`? 
- Trying to be `float`?
- Deliberately trying to support expression input in text form(ie, `1+2*3`?

Whichever this it, it's a bad idea for two reasons:
- Pushing floats through string form is inefficient and precision losing
- You probably don't want to execute arbitrary python code passed in from a config file

**Comment by vrabaud** [comment]
Well, I don't know that Python style. That's a question to ask @mikeferguson  :)  Thx

**Comment by mikeferguson** [comment]
This actually predates me -- I just moved things around and tried to parameterize and clean up -- in particular I think this is for lines like: https://github.com/PR2/pr2_calibration/blob/hydro-devel/pr2_calibration_launch/estimate_params/config_pr2_beta/system.yaml#L109 where it has to evaluate "pi/2"

**Comment by vrabaud** [comment]
good catch ! That .yaml is a custom format right ? Maybe we can just parse pi properly and that''s it.
eval can be a security issue like https://github.com/ros-perception/vision_opencv/pull/112

**Comment by eric-wieser** [comment]
Is `pi` the only global that should be accessible in the expression? Or should we expose all of `math.*`?  Right now, there are way more constants accesible than makes sense. Something like this would work:

https://stackoverflow.com/questions/2371436/evaluating-a-mathematical-expression-in-a-string
